ListUtil.xs

```diff
@@ -1357,75 +1348,98 @@ CODE:
                 SvNV(arg); /* SvIV() sets SVf_IOK even on floats on 5.6 */
 #endif
             }
-#if NVSIZE > IVSIZE  /* open nvsize>ivsize block */
-
-        /* We can detect duplicates by looking *
-         * solely at the value of SvNV(arg)    */
-              
-        nv_arg = SvNV(arg);
-
-        if(nv_arg == 0) {
-            /* use 0 for both 0 and -0.0 */
-            sv_setpvs(keysv, "0");
-        }
-        else if (nv_arg != nv_arg) {
-            /* for NaN, use the platform's normal stringification */
-            sv_setpvf(keysv, "%" NVgf, nv_arg);
+#if NVSIZE > IVSIZE                          /* $Config{nvsize} > $Config{ivsize} */
+        /* Avoid altering arg's flags */ 
+        if(SvUOK(arg))      nv_arg = (NV)SvUV(arg);
+        else if(SvIOK(arg)) nv_arg = (NV)SvIV(arg);
+        else                nv_arg = SvNV(arg);
```

Comment:
This part slightly worries me. Won't it get confused by an IV+NV dualvar, such as those created by:

```perl
# uniqnum not confused by IV'ified floats
{
  my @nums = ( 2.1, 2.2, 2.3 ); # all 2 as IVs
  map { sprintf "%d", $_ } @nums;
  # All @nums now have both NOK and IOK
  is( scalar( uniqnum @nums ), 3, 'uniqnum not confused by dual IV+NV' );
}
```

I'm currently building myself a `-Duselongdouble` perl so I can test on a perl where `NVSIZE > IVSIZE` to see for myself...

Reply:
Update: Seems like this is fine. The SVs gain the private pIOK flag but not the (public) IOK one, so the `uniqnum` code isn't confused:

```
SV = PVNV(0x5600c853bbd0) at 0x5600c8c59ba0
  REFCNT = 2
  FLAGS = (NOK,pIOK,pNOK)
  IV = 2
  NV = 2.1
  PV = 0
SV = PVNV(0x5600c853bba0) at 0x5600c8c5a438
  REFCNT = 2
  FLAGS = (NOK,pIOK,pNOK)
  IV = 2
  NV = 2.2
  PV = 0
SV = PVNV(0x5600c853bb70) at 0x5600c8c59cd8
  REFCNT = 2
  FLAGS = (NOK,pIOK,pNOK)
  IV = 2
  NV = 2.3
  PV = 0
ok 33 - uniqnum not confused by dual IV+NV
```

Reply:
Ah turns out on Perl 5.6 that doesn't hold. It fails:

https://travis-ci.org/github/Dual-Life/Scalar-List-Utils/jobs/672183461

Reply:
The same test fails in the same way with List-Util-1.53.
This looks to me like a bug in the way that perl-5.6.2 is handling dualvars.
I would think dwimmery dictates that
```
map { sprintf "%d", $_ } @nums;
```
should still leave the elements of \@nums being regarded primarily as NVs.
But, if all 4 of IOK , pIOK, NOK and pNOK are set (as is the case with 5.6.2), then who is to say which value to consider ?

Interestingly, on 5.6.2, print() still outputs the NV values of @nums.
I guess that print() must be searching for NOK before it goes for looking for IOK.
Whereas uniqnum() has always looked for IOK first.

I think we should skip that test for any perl that sets dualvars so ambiguously.

Cheers,
Rob